pkg/api/configsync/v1beta1/resource_override.go

```diff
@@ -89,3 +96,17 @@ type ContainerResourcesSpec struct {
 	// +optional
 	MemoryLimit resource.Quantity `json:"memoryLimit,omitempty"`
 }
+
+// ContainerLogLevelOverride allows to override the log level for a container
+type ContainerLogLevelOverride struct {
+	// containerName specifies the name of a reconciler deployment container whose log level will be overridden.
+	// Must be one of the following: "reconciler", "git-sync", "hydration-controller", "oci-sync", or "helm-sync".
+	//
+	// +kubebuilder:validation:Pattern=^(reconciler|git-sync|hydration-controller|oci-sync|helm-sync)$
+	// +optional
+	ContainerName string `json:"containerName,omitempty"`
```

Comment:
Is there any guarantee that the containerName is unique in the array? What happens if the same container is specified multiple times?

Reply:
listType: map and listMergeKey: containerName should enforce uniqueness and help SSA merging

Reply:
thanks for the reccommendation! Changed to use listType: map and listMapKey: containerName. Confirmed that non unique container name will cause CRD validation error now